sdk/testutil/retry/retry_test.go

```diff
@@ -65,6 +116,7 @@ func TestRunWith(t *testing.T) {
 			r.Fatalf("not yet")
 		})
 
+		// TODO: these should all be assert
```

Comment:
Why assert? Do you want the test to keep running if there's a failure in the test suite?

Reply:
The way I see it, you should use `assert` when those assertions are independent, but `require` if they are dependent. Or put another way, use `require` when the following assertions couldn't possibly pass, or are impossible to test.

Overusing `require` means that if all the following assertions fail for independent reasons, we'd need to do multiple fix-and-test passes to identify that that is the case.

Reply:
Gotcha. Thanks for clarifying